src/mina-signer/src/rosetta.ts

```diff
@@ -49,12 +85,93 @@ function fieldFromHex<T extends Field | Scalar>(
+    let payment = unsignedTransactionPaymentToPaymentJson(transaction.payment);
+    signature = signPayment(payment, privateKey, network);
+  }
+  else if (transaction.stakeDelegation !== null) {
+    let delegation = unsignedTransactionStakeDelegationToDelegationJson(transaction.stakeDelegation);
+    signature = signStakeDelegation(delegation, privateKey, network);
+  }
+  else {
+    throw Error('signTransaction: Unsupported transaction');
+  }
+  let publicKey = PublicKey.toBase58(PrivateKey.toPublicKey(PrivateKey.fromBase58(privateKey)));
+  return {
+    data: transaction,
+    signature: signatureJsonToHex(signature),
+    publicKey,
+  };
+}
+
+function unsignedTransactionPaymentToPaymentJson(payment: Payment): PaymentJson {
+  return {
+    common: {
+      fee: payment.fee,
+      feePayer: payment.from,
+      nonce: payment.nonce,
+      validUntil: payment.valid_until ?? defaultValidUntil,
+      memo: payment.memo ?? '',
+    },
+    body: {
+      receiver: payment.to,
+      amount: payment.amount,
+    }
+  }
+}
+
+function unsignedTransactionStakeDelegationToDelegationJson(delegation: StakeDelegation): DelegationJson {
+  return {
+    common: {
+      feePayer: delegation.delegator,
+      fee: delegation.fee,
+      validUntil: delegation.valid_until ?? defaultValidUntil,
+      memo: delegation.memo ?? '',
+      nonce: delegation.nonce,
+    },
+    body: {
+      newDelegate: delegation.new_delegate,
+    }
+  }
+}
+
+function verifyTransaction(signedTransaction: SignedRosetta<UnsignedTransaction>, network: NetworkId): boolean {
+  if (signedTransaction.data.payment !== null) {
+    return verifyPayment(unsignedTransactionPaymentToPaymentJson(signedTransaction.data.payment), signatureJsonFromHex(signedTransaction.signature), signedTransaction.publicKey, network);
+  } else if (signedTransaction.data.stakeDelegation !== null) {
+    return verifyStakeDelegation(unsignedTransactionStakeDelegationToDelegationJson(signedTransaction.data.stakeDelegation), signatureJsonFromHex(signedTransaction.signature), signedTransaction.publicKey, network);
+  }
+  throw Error('verifyTransaction: Unsupported transaction');
+}
+
+// create a signature for /construction/combine payload
+function rosettaCombineSignature(signature: SignedRosetta<UnsignedTransaction>, signingPayload: any): RosettaSignature {
+  let publicKey = PublicKey.fromBase58(signature.publicKey);
+  return {
+    hex_bytes: signature.signature,
+    public_key: {
+      hex_bytes: publicKeyToHex(publicKey),
+      curve_type: "pallas"
+    },
+    signature_type: "schnorr_poseidon",
+    signing_payload: signingPayload
+  }
+}
+
+// create a payload for /construction/combine
+function rosettaCombinePayload(unsignedPayload: UnsignedPayload, privateKey: Json.PrivateKey, network: NetworkId) {
+  let signature = signTransaction(unsignedPayload.unsigned_transaction, privateKey, network);
+  let signatures = [rosettaCombineSignature(signature, unsignedPayload.payloads[0])];
```

Comment:
I don't know how this is supposed to work. but it confuses and worries me that we include a "payload" with the signature which is not related (at least within this function) to what is signed, in any way. You sign `unsigned_transaction` but then include `payloads[0]` in the `RosettaSignature`.

Reply:
This simply streamlines the offline part of signing Rosetta transactions, this is, it's taking the output from Rosetta's `/construction/payloads` endpoint (`UnsignedPayload`), producing a signature for it, and bundling everything together to produce the payload expected by the `/contruction/combine` endpoint.

Reply:
ok I'll take that :)